corehq/apps/app_manager/tests/test_suite_inline_search.py

```diff
@@ -419,6 +418,67 @@ def test_inline_search_with_parent_select(self):
 
         self.assertXmlPartialEqual(expected_entry, suite, "./entry[1]")
 
+    @flag_enabled('USH_SEARCH_FILTER')
+    def test_inline_search_with_parent_relationship_parent_select(self):
+        """Inline search module with 'parent select' relationship is 'parent'"""
+        module = self.app.add_module(Module.new_module("Followup2", None))
+        form = self.app.new_form(2, "Untitled Form", None, attachment=get_simple_form("xmlns1.0"))
+        form.requires = 'case'
+        module.case_type = 'case'
+
+        self.module.parent_select.active = True
+        self.module.parent_select.relationship = 'parent'
+        self.module.parent_select.module_id = module.unique_id
+
+        suite = self.app.create_suite()
+
+        expected_entry = f"""
+        <partial>
+          <entry>
+            <form>xmlns1.0</form>
+            <post url="http://localhost:8000/a/test_domain/phone/claim-case/"
+                relevant="count(instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/case_id]) = 0">
+              <data exclude="count(instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/case_id]) != 0"
+                key="case_id" ref="instance('commcaresession')/session/data/case_id"/>
+              <data exclude="count(instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/parent_id]) != 0"
+                key="case_id" ref="instance('commcaresession')/session/data/parent_id"/>
+            </post>
+            <command id="m0-f0">
+              <text>
+                <locale id="forms.m0f0"/>
+              </text>
+            </command>
+            <instance id="casedb" src="jr://instance/casedb"/>
+            <instance id="commcaresession" src="jr://instance/session"/>
+            <instance id="results:inline" src="jr://instance/remote/results:inline"/>
+            <session>
+              <datum id="parent_id" nodeset="instance('casedb')/casedb/case[@case_type='case'][@status='open']"
+                value="./@case_id" detail-select="m2_case_short"/>
+              <query url="http://localhost:8000/a/test_domain/phone/search/123/"
+                storage-instance="{RESULTS_INSTANCE_INLINE}" template="case" default_search="false" dynamic_search="false">
+                <title>
+                  <text>
+                      <locale id="case_search.m0.inputs"/>
+                  </text>
+                </title>
+                <data key="case_type" ref="'case'"/>
+                <prompt key="name">
+                  <display>
+                    <text>
+                      <locale id="search_property.m0.name"/>
+                    </text>
+                  </display>
+                </prompt>
+              </query>
```

Comment:
This should also have an `ancestor-exists` query (or whatever filter is needed to match the 'parent select' config) otherwise the results that are returned may may not be child cases of the parent case.

Reply:
That makes sense. Is this what we need? [cbc5d4c](https://github.com/dimagi/commcare-hq/pull/33738/commits/cbc5d4c5b75f077f069ea787837b081c68a537a6)

Reply:
Wait actually I don't really get it. Within the session block, there's the datum 
```
<datum id="case_id"
 nodeset="instance('{RESULTS_INSTANCE_INLINE}')/results/case[@case_type='case'][@status='open'][active = 'yes'][not(commcare_is_related_case=true())][index/parent=instance('commcaresession')/session/data/parent_id]"
 ```
Doesn't the `index/parent` portion filter such that returned cases will be childs of the parent case? And this is referenced in the post data 
```
<data exclude="count(instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/case_id]) != 0"
 key="case_id" ref="instance('commcaresession')/session/data/case_id"/>
```

Reply:
The `index/parent` filtering only happens after the search results are returned from ES so if we include a bunch of cases that aren't children of the selected parent then we end up filtering them out in FP. In an extreme case, all the search results returned may be filtered out leaving an empty set of results even though there could be child cases in ES.

Reply:
oh i see, that makes sense. I didn't know that this was where the ES filtering vs post ES filtering was defined. Thanks for explaining